.github/workflows/workflow-run-forge.yaml

```diff
@@ -180,6 +180,33 @@ jobs:
         shell: bash
         run: gcloud config set project aptos-forge-gcp-0
 
+      - name: Check for Forge images
+        if: ${{ !inputs.SKIP_JOB }}
+        id: check-forge-images
+        run: |
+          check_image_exists() {
+            local tag=$1
+            local image_name=$2
+            local output_prefix=$3
+            if [[ -n "$tag" ]]; then
+              if docker manifest inspect us-docker.pkg.dev/aptos-registry/docker/${image_name}:${tag} &> /dev/null; then
+                echo "${output_prefix}_EXISTS=true" >> $GITHUB_OUTPUT
+                echo "${output_prefix}_TAG=$tag" >> $GITHUB_OUTPUT
+                echo "✅ $image_name image found for tag: $tag" >> $GITHUB_STEP_SUMMARY
+              else
+                echo "${output_prefix}_EXISTS=false" >> $GITHUB_OUTPUT
+                echo "⚠️ No $image_name image found for tag: $tag. The latest available image will be used instead." >> $GITHUB_STEP_SUMMARY
```

Comment:
There's code around getting the latest image in forge.py that I didn't want to duplicate here. This is mostly to prompt users to use the `CICD: build-images` label when running Forge tests against their PR. Users can still see the actual images being used by checking the workflow logs

The output shows up in the workflow summary https://github.com/aptos-labs/aptos-core/actions/runs/11410691190

Reply:
May be we should think about unifying these logic in a future PR.

Reply:
Yeah I think the logic to determine the images belongs in GHA. We should pull out the image tags to inputs/outputs to make the Forge runs reproducible. Right now if we rerun the workflow, it could result in different images being used